app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -54,21 +54,22 @@
 
     @NonNull
     @Override
-    public Completable querySearch(@NonNull final String query) {
-        return freeSoundApiService.search(get(query))
-                                  .map(DefaultSearchDataModel::toResults)
-                                  .doOnSubscribe(__ -> searchStateOnceAndStream
-                                          .onNext(SearchState.inProgress()))
-                                  .doOnSuccess(this::storeValueAndClearError)
-                                  .doOnError(storeError(query))
-                                  .toCompletable()
-                                  .onErrorComplete();
+    public Completable querySearch(@NonNull final String query,
+                                   @NonNull final Completable afterThis) {
+        return afterThis.doOnSubscribe(__ -> reportInProgress())
```

Comment:
This is possible with some work. It should be done in the `SearchDataModel`, rather than the View so that the SearchState emissions are the source of truth.